src/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -429,6 +429,80 @@ public static bool TrySkipBlock(BinaryReader reader)
 
             return true;
         }
+
+        public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
+        {
+            const int OffsetToFilenameLength = 22; // from the point after the signature
+
+            if (reader.ReadUInt32() != SignatureConstant)
+                return false;
+
+            if (reader.BaseStream.Length < reader.BaseStream.Position + OffsetToFilenameLength)
+                return false;
+
+            uint minimunVersion = reader.ReadUInt16();
+            int dataDescriptorBit = reader.ReadUInt16() & 0x08;
```

Comment:
This should use https://github.com/dotnet/corefx/blob/0eed4042f056b4d9f6f4d1bf2918e234d67a6f88/src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs#L1253.